client/post-editor/editor-html-toolbar/index.jsx

```diff
@@ -170,11 +170,21 @@ export class EditorHtmlToolbar extends Component {
 			previousSelectionEnd + insertedContentLength;
 	}
 
-	// execCommand( 'insertText' ), needed to preserve the undo stack, does not exist in IE11.
-	// Using the previous version of replacing the entire content value instead.
-	updateEditorContent = isIE11
-		? this.updateEditorContentIE11
-		: this.insertEditorContent;
+	updateEditorContent = ( before, newContent, after ) => {
+		if ( isFirefox ) {
+			// In Firefox, execCommand( 'insertText' ), needed to preserve the undo stack,
+			// always moves the cursor to the end of the content.
+			// A workaround involving manually setting the cursor position and inserting the editor content
+			// is needed to put the cursor back to the correct position.
+			this.updateEditorContentFirefox( before, newContent, after );
+		} else if ( isIE11 ) {
+			// execCommand( 'insertText' ), needed to preserve the undo stack, does not exist in IE11.
+			// Using the previous version of replacing the entire content value instead.
+			this.updateEditorContentIE11( before, newContent, after );
+		} else {
+			this.insertEditorContent( before, newContent, after );
+		}
+	}
```

Comment:
now that we have more than two branches we might consider an _early-abort_ pattern instead of a long chain of `if/else`. it's somewhat of a style choice here but that has been our preference in other places

```js
if ( isFirefox ) {
	// In Firefox…
	return this.updateEditorContentFirefox( before, newContent, after );
}

if ( isIE11 ) {
	// execCommand…
	return this.updateEditorContentIE11( before, newContent, after );
}

return this.insertEditorContent( before, newContent, after );
```

also, was the way of assigning functions before not serving us well? I noticed we moved from something like `update = updateIE11` to `update = ( a, b, c ) => updateIE11( a, b, c )`